packages/flutter_tools/lib/src/asset.dart

```diff
@@ -113,8 +113,9 @@ class AssetBundle {
         manifestDescriptor['uses-material-design'];
 
     for (_Asset asset in assetVariants.keys) {
-      assert(asset.assetFileExists);
-      entries[asset.assetEntry] = new DevFSFileContent(asset.assetFile);
+      assert(asset.assetFileExists || assetVariants.isNotEmpty);
```

Comment:
Is this an invariant at this point, or is this the place where you're checking that the input is valid?
Asserts won't always run in flutter_tools, since we sometimes run it in release mode.